stdlib/io/src/main/ballerina/src/io/readable_record_channel.bal

```diff
@@ -28,30 +30,57 @@ public type ReadableTextRecordChannel object {
     # + rs - Record separator (this could be a regex)
     public function __init(ReadableCharacterChannel charChannel, public string fs = "", public string rs = "",
                            public string fmt = "default") {
-        self.init(charChannel, fs, rs, fmt);
+        initReadableTextRecordChannel(self, charChannel, java:fromString(fs), java:fromString(rs), java:fromString(fmt));
     }
 
-    # Initializes delimited record channel.
-    #
-    # + characterChannel - Character channel which will be used for reading/writing records
-    # + fieldSeparator - Field separator which will separate between fields
-    # + recordSeparator - Record separator which will separate between records
-    # + fmt - Format which will be used to represent the type of record i.e csv
-    function init(ReadableCharacterChannel characterChannel, public string fieldSeparator,
-                public string recordSeparator, public string fmt) = external;
-
     # Checks whether there's a record left to be read.
     #
     # + return - True if there's a record left to be read
-    public function hasNext() returns boolean = external;
+    public function hasNext() returns boolean {
+        return hasNextExtern(self);
+    }
 
     # Get next record from the input/output resource.
     #
     # + return - Set of fields included in the record or `Error` if any error occurred
-    public function getNext() returns @tainted string[]|Error = external;
+    public function getNext() returns @tainted string[]|Error {
+        handle[]|Error result = getNextExtern(self);
+        if (result is Error) {
+            return result;
+        } else {
+            string[] records = [];
+            foreach handle v in result {
+                records.push(<string>java:toString(v));
```

Comment:
Shall we create and return an `Error` instead of directly casting this to a string ?

Otherwise if we get `nil` for the string, this will panic.

Reply:
Internally this will never produce a Java null string. So this toString never produces a nill value.